File: frontend/app/types/task.ts
export type TaskStatus = 'todo' | 'in_progress' | 'done';
export type TaskPriority = 'low' | 'medium' | 'high';

export interface Task {
  id?: string;
  title: string;
  description?: string | null;
  status: TaskStatus;
  priority: TaskPriority;
  energy_level?: number | null;
  estimated_time?: number | null;
  urgency_score?: number | null;
  tags?: string[] | null;
  due_date?: string | null;
  user_id?: string;
  project_id?: string | null;
  created_at?: string;
  updated_at?: string;
}

export type CreateTaskInput = Omit<Task, 'id' | 'user_id' | 'created_at' | 'updated_at'> & {
  title: string;
  status: TaskStatus;
  priority: TaskPriority;
};

export interface TaskFilters {
  status?: Task['status'];
  priority?: Task['priority'];
  tags?: string[];
  dueDate?: 'today' | 'week' | 'month' | 'all';
  searchTerm?: string;
  projectId?: string;
}

export interface TaskSuggestion {
  priority: Task['priority'];
  energy_level: number;
  estimated_time: number;
  suggested_tags: string[];
  category: string;
}

File: backend/app/models/task.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class Task(BaseModel):
    id: Optional[str] = None
    title: str
    description: str
    status: str
    priority: str
    energy_level: Optional[int] = None
    estimated_time: Optional[int] = None
    urgency_score: Optional[float] = None
    tags: List[str] = []
    due_date: Optional[datetime] = None
    user_id: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        json_encoders = {
            datetime: lambda v: v.isoformat() if v else None
        }